### PR TITLE
Hook up live tests

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

File: sdk/storage/azblob/test-resources.json [hunk 1]
```diff
@@ -0,0 +1,679 @@
...
+        "type": "string",
+        "value": "[variables('location')]"
+      },
+      "AZURE_STORAGE_ENCRYPTION_SCOPE": {
+        "type": "string",
+        "value": "[variables('primaryEncryptionScopeName')]"
+      }
+    }
+  }
+
```
Comment (benbp) [inline sdk/storage/azblob/test-resources.json:679]: You may not need all these resources, assuming this is copied over from NET. The main impact will just be deployment time locally and in the pipeline, which takes about 13 minutes in the NET pipelines.

Comment (jhendrixMSFT) [reply sdk/storage/azblob/test-resources.json:679]: Good point, will remove unnecessary resources.

File: sdk/storage/azblob/appendblob/client_test.go [hunk 2]
```diff
@@ -28,8 +29,16 @@ import (
 )
 
 func Test(t *testing.T) {
-	suite.Run(t, &AppendBlobRecordedTestsSuite{})
-	//suite.Run(t, &AppendBlobUnrecordedTestsSuite{})
+	recordMode := os.Getenv("AZURE_RECORD_MODE")
```
Comment (chlowell) [inline sdk/storage/azblob/appendblob/client_test.go:32]: Can we use `recording.GetRecordMode()`?

Comment (jhendrixMSFT) [reply sdk/storage/azblob/appendblob/client_test.go]: Good catch, fixed.

File: sdk/storage/azblob/appendblob/client_test.go [hunk 3]
```diff
@@ -28,8 +29,16 @@ import (
...
-	suite.Run(t, &AppendBlobRecordedTestsSuite{})
-	//suite.Run(t, &AppendBlobUnrecordedTestsSuite{})
+	recordMode := recording.GetRecordMode()
+	t.Logf("Running appendblob Tests in %s mode\n", recordMode)
+	if recordMode == "live" {
+		suite.Run(t, &AppendBlobRecordedTestsSuite{})
+		suite.Run(t, &AppendBlobUnrecordedTestsSuite{})
+	} else if recordMode == "playback" {
+		suite.Run(t, &AppendBlobRecordedTestsSuite{})
+	} else if recordMode == "record" {
```
Comment (chlowell) [inline sdk/storage/azblob/appendblob/client_test.go:39]:
```suggestion
	if recordMode == recording.LiveMode {
		suite.Run(t, &AppendBlobRecordedTestsSuite{})
		suite.Run(t, &AppendBlobUnrecordedTestsSuite{})
	} else if recordMode == recording.PlaybackMode {
		suite.Run(t, &AppendBlobRecordedTestsSuite{})
	} else if recordMode == recording.RecordMode {
```

Comment (jhendrixMSFT) [reply sdk/storage/azblob/appendblob/client_test.go]: Fixed (I saw that `recording.GetRecordMode()` returns a string and assumed there weren't constants for this)

File: sdk/storage/azblob/service/client_test.go [hunk 4]
```diff
@@ -680,8 +691,20 @@ func (s *ServiceUnrecordedTestsSuite) TestContainerRestore() {
 
 			if *cont.Deleted && *cont.Name == containerName {
 				contRestored = true
-				_, err = svcClient.RestoreContainer(context.Background(), containerName, *cont.Version, nil)
-				_require.Nil(err)
+				const maxTries = 5
+				for tries := 0; tries < maxTries; tries++ {
+					_, err = svcClient.RestoreContainer(context.Background(), containerName, *cont.Version, nil)
```
Comment (jhendrixMSFT) [inline sdk/storage/azblob/service/client_test.go:696]: This consistently fails on the first run, and the retries don't help.  Rerunning the test passes.

Comment (jhendrixMSFT) [reply sdk/storage/azblob/service/client_test.go]: Fixed, see code comments for details.

**Review by chlowell — APPROVED** [review]

**Pull request merged**